### ISSUE TITLE
Labelprinter

### ISSUE THREAD

**Issue by deralmin16** [Donkie/Spoolman#294]
Hello, I would like to print the labels using a label printer. Would it be possible to export them, preferably as a CSV and adapt them if necessary?

**Comment by TomW1605** [comment]
depending on the printer you have and how it is setup you might be able to already. many usb label printers present themselves to windows as standard printers so if you set the page size right they can print from any software.

i use these settings to print directly to my Brother QL-700 on a continuous 62mm label tape 
![image](https://github.com/Donkie/Spoolman/assets/17092573/a6b198b5-1e31-44dc-a370-1f63b5ca4ec4)

**Comment by vikavorkin** [comment]
Had the same issue using a the PTouch editor.
I have a PR adding CSV export that solved my issue - https://github.com/Donkie/Spoolman/pull/193

**Comment by braintapper** [comment]
Hi @TomW1605 I tried printing with the settings from your screenshot (Windows) on my QL700 but I get an error saying that the label size 29mm x 90mm (which I think was the original default in Spoolman print dialog) doesn't match the printer.

I customized the label size to 62x24 in the Spoolman print dialog and set the paper size to 62mm in the Windows printer dialog for the QL700.

Am I missing something?

Thanks in advance

**Comment by braintapper** [comment]
Hi @TomW1605 I figured it out - here's the resolution, hopefully this helps someone else...  the printer ignores the page size specified in the Windows print modal, I had to change the default paper size for the printer in the Printers control panel. Which is counter intuitive. The printer works perfectly for any print size when using the Brother app irrespective of the global page size setting.

**Comment by TomW1605** [comment]
yeah, the drivers for these printers are not great for working with anything other than the provided apps. but once you get those settings right they work well and you dont have to use the app

**Comment by dave-lang** [comment]
I've done some testing on a Brother D610BT and singular and multiple labels work, although it requires some creative margin settings and the width needs to match what's configured in the printer settings to work correctly - appears to be entirely printer driver related

![image](https://github.com/user-attachments/assets/2a561fff-8a06-49c1-90a2-d106d4a807dd)

![image](https://github.com/user-attachments/assets/55a8b7e7-2815-4f17-96df-c07bbb66954b)

![WIN_20250114_16_34_46_Pro](https://github.com/user-attachments/assets/09d16504-ec57-4cb2-ad3d-1fe87f103df2)

**Comment by akira69** [comment]
3dfilamentprofiles.com has .aml files you can download and open in the labelife app to print to a label printer like the M220.  This is perfect.... How about some AML export options???

**Comment by akira69** [comment]
This all would be addressed by:
https://github.com/Donkie/Spoolman/pull/860 - Separate AML/PNG export flow/definition
https://github.com/Donkie/Spoolman/pull/857 - Awesomer Logos

**Comment by wabudd1** [comment]
Popping in to mention that [this fork of brother-ql-web](https://github.com/DL6ER/brother_ql_web) is being actively developed and works quite well with my Brother QL-810W.  It has an API to send images to and it would be kinda cool if Spoolman could have a feature to send label images to that API.

For now, I've been downloading labels from Spoolman and plopping them in the brother-ql-web label designer one by one.
